pset-iterators.js

```diff
@@ -184,8 +208,24 @@ console.log('Problem 7:')
 
 // Add your code below this line
 
-// Add your code above this line
+const shortTermPresidents = presidents.filter((president) => {
+
+	if (president.left_office === null){
+		president.left_office = "no date specified"
+	}
 
+  const tookOfficeYears = parseInt(president.took_office.split("-")[0])
+	const leftOfficeYears = parseInt(president.left_office.split("-")[0])
+
+  if ((leftOfficeYears - tookOfficeYears) < 4){
+    return true
+  } else {
+    return false
+  }
+
+})
+// Add your code above this line
+console.log(shortTermPresidents)
```

Comment:
@jl1982 Good job here on this assignment using the filter method to check the `tookOfficeYears` and `leftOfficeYears` and subtract the values and return true if less than 4 years. Good job checking to see if the value is null and change the value to a string so your code can run.

Take a look at this alternate solution [here](https://github.com/jsr20190826/hw-03-iterators-solution/blob/96bab20deba3ad116374a5827291a951faf5d555/pset-iterators.js#L201)